be/src/storage/rowset/segment_v2/array_column_iterator.cpp

```diff
@@ -122,6 +122,64 @@ Status ArrayColumnIterator::next_batch(size_t* n, vectorized::Column* dst) {
     return Status::OK();
 }
 
+Status ArrayColumnIterator::next_batch(vectorized::SparseRange& range, vectorized::Column* dst) {
+    vectorized::ArrayColumn* array_column = nullptr;
+    vectorized::NullColumn* null_column = nullptr;
+    if (dst->is_nullable()) {
+        auto* nullable_column = down_cast<vectorized::NullableColumn*>(dst);
+
+        array_column = down_cast<vectorized::ArrayColumn*>(nullable_column->data_column().get());
+        null_column = down_cast<vectorized::NullColumn*>(nullable_column->null_column().get());
+    } else {
+        array_column = down_cast<vectorized::ArrayColumn*>(dst);
+    }
+
+    // 1. Read null column
+    if (_null_iterator != nullptr) {
+        RETURN_IF_ERROR(_null_iterator->next_batch(range, null_column));
+        down_cast<vectorized::NullableColumn*>(dst)->update_has_null();
+    }
```

Comment:
what if `_null_iterator == nullptr` and `null_column != nullptr` ?

Reply:
> what if `_null_iterator == nullptr` and `null_column != nullptr` ?

`_null_iterator == nullptr` means that all values of the column are not null, so reading data from `element_iterator` is enough even the `dst_column` is nullable

Reply:
> > what if `_null_iterator == nullptr` and `null_column != nullptr` ?
> 
> `_null_iterator == nullptr` means that all values of the column are not null, so reading data from `element_iterator` is enough even the `dst_column` is nullable

But you have to update the `null_column`, just reading `element_iterator` is not enough.

Reply:
> 

this should not happen actually, if `_null_iterator` is nullptr, it means `dst_column` is not null.   The `_null_iterator' is constructed  only if the column is nullable

Reply:
> this should not happen actually, if _null_iterator is nullptr, it means dst_column is not null.
This is an implicit assumption, you cannot rely on it. If you do reply on this assumption, make the implicit assumption explicit by adding an assertion at the beginning, like this:
```
CHECK((_null_iterator != nullptr && null_column != nullptr) || (_null_iterator == nullptr && null_column == nullptr));
```

Reply:
> > this should not happen actually, if _null_iterator is nullptr, it means dst_column is not null.
> 
> This is an implicit assumption, you cannot rely on it. If you do rely on this assumption, make the implicit assumption explicit by adding an assertion at the beginning, like this:
> 
> ```
> CHECK((_null_iterator != nullptr && null_column != nullptr) || (_null_iterator == nullptr && null_column == nullptr));
> ```

done